src/NextGen/Gateways/Stripe/NextGenStripeGateway/NextGenStripeGatewaySubscriptionModule.php

```diff
@@ -0,0 +1,379 @@
+use Give\Framework\PaymentGateways\Contracts\Subscription\SubscriptionDashboardLinkable;
+use Give\Framework\PaymentGateways\Exceptions\PaymentGatewayException;
+use Give\Framework\PaymentGateways\SubscriptionModule;
+use Give\Framework\Support\ValueObjects\Money;
+use Give\PaymentGateways\Gateways\Stripe\Traits\CanSetupStripeApp;
+use Give\Subscriptions\Models\Subscription;
+use Give\Subscriptions\ValueObjects\SubscriptionMode;
+use Give\Subscriptions\ValueObjects\SubscriptionStatus;
+use GiveRecurring\Infrastructure\Exceptions\PaymentGateways\Stripe\UnableToCreateStripePlan;
+use GiveRecurring\PaymentGateways\DataTransferObjects\SubscriptionDto;
+use GiveRecurring\PaymentGateways\Stripe\Actions\RetrieveOrCreatePlan;
+use Stripe\Customer;
+use Stripe\Exception\ApiErrorException;
+use Stripe\PaymentMethod;
+use Stripe\Plan;
+use Stripe\Subscription as StripeSubscription;
+
+/**
+ * @unreleased
+ */
+class NextGenStripeGatewaySubscriptionModule extends SubscriptionModule implements SubscriptionDashboardLinkable,
+                                                                                   SubscriptionAmountEditable
+{
+    use CanSetupStripeApp;
+    use NextGenStripeRepository;
+
+    /**
+     * @unreleased
+     *
+     * @throws Exception
+     * @throws ApiErrorException
+     * @throws \Exception
+     */
+    public function createSubscription(
+        Donation $donation,
+        Subscription $subscription,
+        $gatewayData
+    ): GatewayCommand {
+        /**
+         * Initialize the Stripe SDK using Stripe::setAppInfo()
+         */
+        $this->setUpStripeAppInfo($donation->formId);
+
+        /**
+         * Get data from client request
+         */
+        $stripeGatewayData = StripeGatewayData::fromRequest($gatewayData);
+
+        /**
+         * Get or create a Stripe customer
+         */
+        $customer = $this->getOrCreateStripeCustomerFromDonation(
+            $stripeGatewayData->stripeConnectedAccountId,
+            $donation
+        );
+
+        /**
+         * Setup Stripe Plan
+         */
+        $plan = $this->createStripePlan($subscription);
+
+        /**
+         * Create Stripe Subscription
+         */
+        $stripeSubscription = $this->createStripeSubscription(
+            $donation,
+            $subscription,
+            $customer,
+            $plan
+        );
+
+        /**
+         * Update Subscription Meta
+         */
+        $this->updateSubscriptionMetaFromStripeSubscription(
+            $subscription,
+            $stripeSubscription
+        );
+
+        /**
+         * Update Initial Donation Meta
+         */
+        $this->updateSubscriptionInitialDonationMetaFromStripeSubscription(
+            $stripeSubscription,
+            $donation
+        );
+
+        /**
+         * Return response to client
+         */
+        return new RespondToBrowser([
+            'clientSecret' => $stripeSubscription->latest_invoice->payment_intent->client_secret,
```

Comment:
I'm assuming that this is required by Stripe, but sending a "secret" to the browser is usually a red flag - just confirming that this isn't leaking a secret key.

Reply:
It looks like it might be a single use, or at least limited relevance, secret, but might be worth adding a note as to why we are passing this to the browser.

Reply:
Stripe's `clientSecret` is specific to confirming payments on the client hence the _client_ secret.  Despite the _secret_ wording it's safe to expose to the browser  - they just [say](https://stripe.com/docs/api/payment_intents/object#payment_intent_object-client_secret) not to store or log it.  I added a couple comments b97ff155